doc/boot-requirements.md

```diff
@@ -63,42 +63,49 @@
 			- **only requires to use the existing EFI partition**
 	- with a LVM-based proposal
 		- if there are no EFI partitions
-			- **requires /boot and a new /boot/efi partition**
+			- **requires only a new /boot/efi partition**
 		- if there is already an EFI partition
-			- **requires /boot and a reused /boot/efi partition**
+			- **only requires to use the existing EFI partition**
 - not using UEFI (legacy PC)
-	- with an MS-DOS partition table
-		- with sufficently large MBR gap
+	- with GPT partition table
+		- in a partitions-based proposal
+			- if there is no GRUB partition
+				- **requires a new GRUB partition**
+			- if there is already a GRUB partition
+				- **does not require any particular volume**
+		- in a LVM-based proposal
+			- if there is no GRUB partition
+				- **requires a new GRUB partition**
+			- if there is already a GRUB partition
+				- **does not require any particular volume**
+	- with a MS-DOS partition table
+		- if the MBR gap is big enough to embed Grub
 			- in a partitions-based proposal
 				- **does not require any particular volume**
 			- in a LVM-based proposal
-				- **requires only a /boot partition**
+				- if the MBR gap has additional space for grubenv
+					- **does not require any particular volume**
+				- if the MBR gap has no additional space
+					- **requires only a /boot partition**
 		- with too small MBR gap
-			- **raises an exception**
-		- with no MBR gap
-			- **raises an exception**
-		- with GPT partition table
-			- if there is no GRUB partition
-				- in a partitions-based proposal
-					- **only requires a new GRUB partition**
-				- in a LVM-based proposal
-					- **requires /boot and a GRUB partitions**
-			- if there is already a GRUB partition
-				- in a partitions-based proposal
+			- in a partitions-based proposal
+				- if proposing root (/) as Btrfs
```

Comment:
Just tested and it did not work. btrfs has only 64k embedding space which is not enough for grub.
Maybe grub has advanced meanwhile - but please check back with mchang.